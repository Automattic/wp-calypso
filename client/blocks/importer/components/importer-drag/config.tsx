import { translate, TranslateOptions, TranslateOptionsText } from 'i18n-calypso';
import React from 'react';
import { convertPlatformName } from 'calypso/blocks/import/util';
import importerConfig, { ImporterConfig } from 'calypso/lib/importer/importer-config';
import SupportLink from '../support-link';
import type { Importer } from 'calypso/blocks/importer/types';

const translateConfig: { [ key: string ]: Partial< ImporterConfig > } = {
	blogger: {
		description:
			'Import posts, pages, comments, tags, and images from a %(importerName)s export file.',
		uploadDescription:
			'A %(importerName)s export is ' +
			'an XML file with your page and post content, or a zip archive ' +
			'containing several XML files. ' +
			'{{supportLink/}}',
	},
	medium: {
		description:
			'Import your posts, tags, images, and videos from your %(importerName)s export file',
		uploadDescription:
			'A %(importerName)s export file is a ZIP file containing several HTML files with your stories. ' +
			'{{supportLink/}}',
	},
	squarespace: {
		description:
			'Import posts, pages, comments, tags, and images from a %(importerName)s export file.',
		uploadDescription:
			'A %(importerName)s export is ' +
			'an XML file with your page and post content, or a zip archive ' +
			'containing several XML files. ' +
			'{{supportLink/}}',
	},
	wordpress: {
		description: 'Import posts, pages, and media from your %(importerName)s export file.',
		uploadDescription:
			'A %(importerName)s export is ' +
			'an XML file with your page and post content, or a zip archive ' +
			'containing several XML files. ' +
			'{{supportLink/}}',
	},
};

export function getImportDragConfig( importer: Importer, supportLinkModal?: boolean ) {
	const importerData = importerConfig()[ importer ];

	const options: TranslateOptions = {
		args: {
			importerName: convertPlatformName( importer ),
		},
		components: {
			supportLink: <SupportLink importer={ importer } supportLinkModal={ supportLinkModal } />,
		},
	};

	importerData.title = translate( 'Import content from %(importerName)s', {
		...options,
		textOnly: true,
	} as TranslateOptionsText ) as string;

	importerData.description = translate(
		// eslint-disable-next-line wpcalypso/i18n-no-variables
		translateConfig[ importer ].description as string,
		options
	);

	importerData.uploadDescription = translate(
		// eslint-disable-next-line wpcalypso/i18n-no-variables
		translateConfig[ importer ].uploadDescription as string,
		options
	);

	return importerData;
}
