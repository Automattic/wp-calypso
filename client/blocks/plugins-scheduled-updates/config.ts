export const MAX_SELECTABLE_PLUGINS = 10;
export const MAX_SELECTABLE_PATHS = 5;
