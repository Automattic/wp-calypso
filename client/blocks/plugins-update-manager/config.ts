export const MAX_SCHEDULES = 2;
export const MAX_SELECTABLE_PLUGINS = 10;
