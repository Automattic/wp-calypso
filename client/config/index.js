/**
 * External dependencies
 */
import cookie from 'cookie';

/**
 * Internal dependencies
 */

import createConfig from 'calypso/lib/create-config';

/**
 * Manages config flags for various deployment builds
 *
 * @module config/index
 */
if ( 'undefined' === typeof window || ! window.configData ) {
	throw new ReferenceError(
		'No configuration was found: please see client/config/README.md for more information'
	);
}

const configData = window.configData;

// calypso.live matches
// hash-abcd1234.calypso.live matches
// calypso.live.com doesn't match
const CALYPSO_LIVE_REGEX = /^([a-zA-Z0-9-]+\.)?calypso\.live$/;

// check if the current browser location is *.calypso.live
export function isCalypsoLive() {
	return typeof window !== 'undefined' && CALYPSO_LIVE_REGEX.test( window.location.host );
}

function applyFlags( flagsString, modificationMethod ) {
	const flags = flagsString.split( ',' );
	flags.forEach( ( flagRaw ) => {
		const flag = flagRaw.replace( /^[-+]/, '' );
		const enabled = ! /^-/.test( flagRaw );
		configData.features[ flag ] = enabled;
		// eslint-disable-next-line no-console
		console.log(
			'%cConfig flag %s via %s: %s',
			'font-weight: bold;',
			enabled ? 'enabled' : 'disabled',
			modificationMethod,
			flag
		);
	} );
}

const flagEnvironments = [ 'wpcalypso', 'horizon', 'stage', 'jetpack-cloud-stage' ];

if (
	process.env.NODE_ENV === 'development' ||
	flagEnvironments.includes( configData.env_id ) ||
	isCalypsoLive()
) {
	const cookies = cookie.parse( document.cookie );

	if ( cookies.flags ) {
		applyFlags( cookies.flags, 'cookie' );
	}

	const match =
		document.location.search && document.location.search.match( /[?&]flags=([^&]+)(&|$)/ );
	if ( match ) {
		applyFlags( match[ 1 ], 'URL' );
	}
}

const configApi = createConfig( configData );

export default configApi;
export const isEnabled = configApi.isEnabled;
