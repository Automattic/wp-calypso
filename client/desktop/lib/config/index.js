const pkg = require( '../../../package.json' );
const config = require( '../../config.json' );

// Merge in some details from package.json
config.name = pkg.productName;
config.description = 'WordPress Desktop';
config.version = pkg.version;
config.author = pkg.author;

config.isRelease = function () {
	return this.build === 'release';
};

config.isUpdater = function () {
	return this.build === 'updater';
};

config.isBeta = function () {
	return this.build === 'beta';
};

module.exports = config;
