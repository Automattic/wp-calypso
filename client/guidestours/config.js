/** @ssr-ready **/

/**
 * External dependencies
 */
import React from 'react';

/**
 * Internal dependencies
 */
import i18n from 'lib/mixins/i18n';

function get() {
	return {
		init: {
			text: i18n.translate( "{{strong}}Need a hand?{{/strong}} We'd love to show you around the place, and give you some ideas for what to do next.", {
				components: {
					strong: <strong />,
				}
			} ),
			type: 'GuidesFirstStep',
			placement: 'right',
			next: 'my-sites',
		},
		'my-sites': {
			target: 'my-sites',
			type: 'GuidesActionStep',
			icon: 'my-sites',
			placement: 'below',
			text: i18n.translate( "{{strong}}First things first.{{/strong}} Up here, you'll find tools for managing your site's content and design.", {
				components: {
					strong: <strong />,
				}
			} ),
			next: 'sidebar',
		},
		sidebar: {
			text: i18n.translate( 'The sidebar menu lets you navigate around, and will adapt to give you the tools you need when you need them.' ),
			type: 'GuidesBasicStep',
			target: 'sidebar',
			placement: 'beside',
			next: 'themes',
		},
		themes: {
			text: i18n.translate( "Change your {{strong}}Theme{{/strong}} to choose a new layout, or {{strong}}Customize{{/strong}} your theme's colors, font, and more.", {
				components: {
					strong: <strong />,
				}
			} ),
			type: 'GuidesBasicStep',
			target: 'themes',
			placement: 'below',
			next: 'finish',
		},
		finish: {
			placement: 'center',
			text: i18n.translate( "{{strong}}That's it!{{/strong}} Now that you know a few of the basics, feel free to wander around.", {
				components: {
					strong: <strong />,
				}
			} ),
			type: 'GuidesFinishStep',
			linkLabel: i18n.translate( 'Get the Most from WordPress.com' ),
			linkUrl: 'https://learn.wordpress.com',
		}
	};
}

export default {
	get,
	version: '20160421',
};
