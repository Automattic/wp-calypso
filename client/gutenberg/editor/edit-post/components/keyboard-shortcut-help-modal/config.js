/** @format */
/* eslint-disable wpcalypso/import-docblock */
/**
 * WordPress dependencies
 */
import { displayShortcutList, shortcutAriaLabel } from '@wordpress/keycodes';
import { __ } from '@wordpress/i18n';

const {
	// Cmd+<key> on a mac, Ctrl+<key> elsewhere.
	primary,
	// Shift+Cmd+<key> on a mac, Ctrl+Shift+<key> elsewhere.
	primaryShift,
	// Option+Cmd+<key> on a mac, Ctrl+Alt+<key> elsewhere.
	primaryAlt,
	// Shift+Alt+Cmd+<key> on a mac, Ctrl+Shift+Akt+<key> elsewhere.
	secondary,
	// Ctrl+Alt+<key> on a mac, Shift+Alt+<key> elsewhere.
	access,
	ctrl,
	alt,
	ctrlShift,
	shiftAlt,
} = displayShortcutList;

const globalShortcuts = {
	title: __( 'Global shortcuts' ),
	shortcuts: [
		{
			keyCombination: access( 'h' ),
			description: __( 'Display this help.' ),
		},
		{
			keyCombination: primary( 's' ),
			description: __( 'Save your changes.' ),
		},
		{
			keyCombination: primary( 'z' ),
			description: __( 'Undo your last changes.' ),
		},
		{
			keyCombination: primaryShift( 'z' ),
			description: __( 'Redo your last undo.' ),
		},
		{
			keyCombination: primaryShift( ',' ),
			description: __( 'Show or hide the settings sidebar.' ),
			ariaLabel: shortcutAriaLabel.primaryShift( ',' ),
		},
		{
			keyCombination: access( 'o' ),
			description: __( 'Open the block navigation menu.' ),
		},
		{
			keyCombination: ctrl( '`' ),
			description: __( 'Navigate to the next part of the editor.' ),
			ariaLabel: shortcutAriaLabel.ctrl( '`' ),
		},
		{
			keyCombination: ctrlShift( '`' ),
			description: __( 'Navigate to the previous part of the editor.' ),
			ariaLabel: shortcutAriaLabel.ctrlShift( '`' ),
		},
		{
			keyCombination: shiftAlt( 'n' ),
			description: __( 'Navigate to the next part of the editor (alternative).' ),
		},
		{
			keyCombination: shiftAlt( 'p' ),
			description: __( 'Navigate to the previous part of the editor (alternative).' ),
		},
		{
			keyCombination: alt( 'F10' ),
			description: __( 'Navigate to the nearest toolbar.' ),
		},
		{
			keyCombination: secondary( 'm' ),
			description: __( 'Switch between Visual Editor and Code Editor.' ),
		},
	],
};

const selectionShortcuts = {
	title: __( 'Selection shortcuts' ),
	shortcuts: [
		{
			keyCombination: primary( 'a' ),
			description: __( 'Select all text when typing. Press again to select all blocks.' ),
		},
		{
			keyCombination: 'Esc',
			description: __( 'Clear selection.' ),
			/* translators: The 'escape' key on a keyboard. */
			ariaLabel: __( 'Escape' ),
		},
	],
};

const blockShortcuts = {
	title: __( 'Block shortcuts' ),
	shortcuts: [
		{
			keyCombination: primaryShift( 'd' ),
			description: __( 'Duplicate the selected block(s).' ),
		},
		{
			keyCombination: access( 'z' ),
			description: __( 'Remove the selected block(s).' ),
		},
		{
			keyCombination: primaryAlt( 't' ),
			description: __( 'Insert a new block before the selected block(s).' ),
		},
		{
			keyCombination: primaryAlt( 'y' ),
			description: __( 'Insert a new block after the selected block(s).' ),
		},
		{
			keyCombination: '/',
			description: __( 'Change the block type after adding a new paragraph.' ),
			/* translators: The forward-slash character. e.g. '/'. */
			ariaLabel: __( 'Forward-slash' ),
		},
	],
};

const textFormattingShortcuts = {
	title: __( 'Text formatting' ),
	shortcuts: [
		{
			keyCombination: primary( 'b' ),
			description: __( 'Make the selected text bold.' ),
		},
		{
			keyCombination: primary( 'i' ),
			description: __( 'Make the selected text italic.' ),
		},
		{
			keyCombination: primary( 'u' ),
			description: __( 'Underline the selected text.' ),
		},
		{
			keyCombination: primary( 'k' ),
			description: __( 'Convert the selected text into a link.' ),
		},
		{
			keyCombination: primaryShift( 'k' ),
			description: __( 'Remove a link.' ),
		},
		{
			keyCombination: access( 'd' ),
			description: __( 'Add a strikethrough to the selected text.' ),
		},
		{
			keyCombination: access( 'x' ),
			description: __( 'Display the selected text in a monospaced font.' ),
		},
	],
};

export default [ globalShortcuts, selectionShortcuts, blockShortcuts, textFormattingShortcuts ];
