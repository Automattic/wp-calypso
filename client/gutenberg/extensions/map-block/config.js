/**
 * Wordpress dependencies
 */

import { __ } from '@wordpress/i18n';

/**
 * External dependencies
 */

 /**
 * Internal dependencies
 */

export const CONFIG = {
	name: 'atavist/maps',
	title: __( 'Map' ),
	icon: <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M20.5 3l-.16.03L15 5.1 9 3 3.36 4.9c-.21.07-.36.25-.36.48V20.5c0 .28.22.5.5.5l.16-.03L9 18.9l6 2.1 5.64-1.9c.21-.07.36-.25.36-.48V3.5c0-.28-.22-.5-.5-.5zM15 19l-6-2.11V5l6 2.11V19z"/><path d="M0 0h24v24H0z" fill="none"/></svg>,
	category: 'common',
	keywords: [
		__( 'Map' ),
		__( 'Atavist' )
	],
	attributes: {
		align: {
			type: 'string'
		},
		the_caption: {
			source: 'text',
			selector: '.atavist-caption'
		},
		points: {
			type: 'array',
			default: []
		},
		map_style: {
			type: 'string',
			default: 'default'
		},
		zoom: {
			type: 'integer',
			default: 13
		},
		map_center: {
			type: 'object',
			default: {
				latitude: 37.7749295,
				longitude: -122.41941550000001
			}
		},
		marker_color: {
			type: 'string',
			default: 'red'
		},
		api_key: {
			type: 'string'
		}
	},
	styles: {
		default: {
			map_type: 'ROADMAP',
			styles: [
				{
					elementType: 'labels',
					stylers: [ {
						visibility: 'on'
					} ]
				},
				{
					featureType: 'poi',
					elementType: 'labels.text',
					stylers: [ {
						visibility: 'simplified'
					} ]
				},
				{
					featureType: 'poi',
					elementType: 'labels.icon',
					stylers: [ {
						visibility: 'off'
					} ]
				},
				{
					featureType: 'transit',
					elementType: 'labels.icons',
					stylers: [ {
						visibility: 'off'
					} ]
				},
				{
					featureType: 'transit',
					elementType: 'labels.text',
					stylers: [ {
						visibility: 'off'
					} ]
				}
			]
		},
		satellite: {
			map_type: 'SATELLITE',
			styles: [
				{
					elementType: 'labels',
					stylers: [ {
						visibility: 'on'
					} ]
				},
				{
					featureType: 'poi',
					elementType: 'labels.text',
					stylers: [ {
						visibility: 'simplified'
					} ]
				},
				{
					featureType: 'poi',
					elementType: 'labels.icon',
					stylers: [ {
						visibility: 'off'
					} ]
				},
				{
					featureType: 'transit',
					elementType: 'labels.icons',
					stylers: [ {
						visibility: 'off'
					} ]
				},
				{
					featureType: 'transit',
					elementType: 'labels.text',
					stylers: [ {
						visibility: 'off'
					} ]
				}
			]
		},
		satellite_with_features: {
			map_type: 'HYBRID',
			styles: [
				{
					elementType: 'labels',
					stylers: [ {
						visibility: 'on'
					} ]
				},
				{
					featureType: 'poi',
					elementType: 'labels.text',
					stylers: [ {
						visibility: 'simplified'
					} ]
				},
				{
					featureType: 'poi',
					elementType: 'labels.icon',
					stylers: [ {
						visibility: 'off'
					} ]
				},
				{
					featureType: 'transit',
					elementType: 'labels.icons',
					stylers: [ {
						visibility: 'off'
					} ]
				},
				{
					featureType: 'transit',
					elementType: 'labels.text',
					stylers: [ {
						visibility: 'off'
					} ]
				}
			]
		},
		terrain: {
			map_type: 'TERRAIN',
			styles: [
				{
					featureType: 'administrative',
					stylers: [ {
						visibility: 'off'
					} ]
				},
				{
					elementType: 'labels',
					stylers: [ {
						visibility: 'off'
					} ]
				},
				{
					featureType: 'poi',
					stylers: [ {
						visibility: 'simplified'
					} ]
				},
				{
					featureType: 'poi',
					elementType: 'labels',
					stylers: [ {
						visibility: 'off'
					} ]
				},
				{
					featureType: 'road',
					stylers: [ {
						visibility: 'off'
					} ]
				},
				{
					featureType: 'transit',
					stylers: [ {
						visibility: 'off'
					} ]
				},
				{
					featureType: 'water',
					elementType: 'labels',
					stylers: [ {
						visibility: 'off'
					} ]
				}
			]
		},
		terrain_with_features: {
			map_type: 'TERRAIN',
			styles: [
				{
					elementType: 'labels',
					stylers: [ {
						visibility: 'on'
					} ]
				},
				{
					featureType: 'poi',
					elementType: 'labels.text',
					stylers: [ {
						visibility: 'simplified'
					} ]
				},
				{
					featureType: 'poi',
					elementType: 'labels.icon',
					stylers: [ {
						visibility: 'off'
					} ]
				},
				{
					featureType: 'transit',
					elementType: 'labels.icons',
					stylers: [ {
						visibility: 'off'
					} ]
				},
				{
					featureType: 'transit',
					elementType: 'labels.text',
					stylers: [ {
						visibility: 'off'
					} ]
				}
			]
		},
		black_and_white: {
			map_type: 'ROADMAP',
			styles: [
				{
					stylers: [ {
						saturation: -100
					} ]
				},
				{
					elementType: 'labels',
					stylers: [ {
						visibility: 'on'
					} ]
				},
				{
					featureType: 'poi',
					elementType: 'labels.text',
					stylers: [ {
						visibility: 'simplified'
					} ]
				},
				{
					featureType: 'poi',
					elementType: 'labels.icon',
					stylers: [ {
						visibility: 'off'
					} ]
				},
				{
					featureType: 'transit',
					elementType: 'labels.text',
					stylers: [ {
						visibility: 'simplified'
					} ]
				}
			]
		}
	},
	map_styleOptions: [
		{
			value: 'default',
			label: 'Basic'
		},
		{
			value: 'black_and_white',
			label: 'Black and white'
		},
		{
			value: 'satellite',
			label: 'Satellite'
		},
		{
			value: 'satellite_with_features',
			label: 'Satellite (with features)'
		},
		{
			value: 'terrain',
			label: 'Terrain'
		},
		{
			value: 'terrain_with_features',
			label: 'Terrain (with features)'
		}
	],
	GOOGLE_MAPS_API_KEY: 'AIzaSyDaj7klnWKpzGx0W5PonA73Dgr68Me8cyg',
	baseClasses: [
		'atavist-block',
		'atavist-simple-map'
	],
	validAlignments: [
		'left',
		'center',
		'right',
		'wide',
		'full'
	],
	markerIcon: <svg width='14' height='20' viewBox='0 0 14 20' xmlns='http://www.w3.org/2000/svg'><g id='Page-1' fill='none' fillRule='evenodd'><g id='outline-add_location-24px' transform='translate(-5 -2)'><polygon id='Shape' points='0 0 24 0 24 24 0 24' /><path d='M12,2 C8.14,2 5,5.14 5,9 C5,14.25 12,22 12,22 C12,22 19,14.25 19,9 C19,5.14 15.86,2 12,2 Z M7,9 C7,6.24 9.24,4 12,4 C14.76,4 17,6.24 17,9 C17,11.88 14.12,16.19 12,18.88 C9.92,16.21 7,11.85 7,9 Z M13,6 L11,6 L11,8 L9,8 L9,10 L11,10 L11,12 L13,12 L13,10 L15,10 L15,8 L13,8 L13,6 Z' id='Shape' fill='#000' fillRule='nonzero' /></g></g></svg>
};
