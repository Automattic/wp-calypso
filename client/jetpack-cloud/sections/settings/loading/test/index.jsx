/**
 * @jest-environment jsdom
 */

/**
 * External dependencies
 */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';

/**
 * Internal dependencies
 */
import { render } from 'config/testing-library';
import AdvancedCredentialsLoadingPlaceholder from '../index';

describe( 'AdvancedCredentialsLoadingPlaceholder', () => {
	it( 'should render correctly', () => {
		const { container } = render( <AdvancedCredentialsLoadingPlaceholder /> );

		expect( container ).toMatchSnapshot();
	} );
} );
