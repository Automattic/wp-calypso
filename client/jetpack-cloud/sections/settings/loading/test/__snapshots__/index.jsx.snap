// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvancedCredentialsLoadingPlaceholder should render correctly 1`] = `
<div>
  <div
    class="loading"
  >
    <div
      class="loading__placeholder"
    />
  </div>
</div>
`;
